docs/keywords.txt

```diff
@@ -0,0 +1,294 @@
+
+``description``
+    A string describing the package in a single line.
+
+``long_description``
+    A string providing a longer description of the package.
+
+``long_description_content_type``
+    A string specifying the content type is used for the ``long_description``
+    (e.g. ``text/markdown``)
+
+``author``
+    A string specifying the author of the package.
+
+``author_email``
+    A string specifying the email address of the package author.
+
+``maintainer``
+    A string specifying the name of the current maintainer, if different from
+    the author. Note that if the maintainer is provided, setuptools will use it
+    as the author in ``PKG-INFO``.
+
+``maintainer_email``
+    A string specifying the email address of the current maintainer, if
+    different from the author.
+
+``url``
+    A string specifying the URL for the package homepage.
+
+``download_url``
+    A string specifying the URL to download the package.
+
+``packages``
+    A list of strings specifying the packages that setuptools will manipulate.
+
+``py_modules``
+    A list of strings specifying the modules that setuptools will manipulate.
+
+``scripts``
+    A list of strings specifying the standalone script files to be built and
+    installed.
+
+``ext_package``
+    A string specifying the base package name for the extensions provided by
+    this package.
+
+``ext_modules``
+    A list of instances of ``setuptools.Extension`` providing the list of
+    Python extensions to be built.
+
+``classifiers``
+    A list of strings describing the categories for the package.
+
+``distclass``
+    A subclass of ``Distribution`` to use.
+
+``script_name``
+    A string specifying the name of the setup.py script -- defaults to
+    ``sys.argv[0]``
+
+``script_args``
+    A list of strings defining the arguments to supply to the setup script.
+
+``options``
+    A dictionary providing the default options for the setup script.
+
+``license``
+    A string specifying the license of the package.
+
+``keywords``
+    A list of strings or a comma-separated string providing descriptive
+    meta-data. See: `PEP 0314`_.
+
+.. _PEP 0314: https://www.python.org/dev/peps/pep-0314/
+
+``platforms``
+    A list of strings or comma-separated string.
+
+``cmdclass``
+    A dictionary providing a mapping of command names to ``Command``
+    subclasses.
+
+``data_files``
+
+    .. warning::
+        ``data_files`` is deprecated. It does not work with wheels, so it
+        should be avoided.
+
+    A list of strings specifying the data files to install.
+
+``package_dir``
+    A dictionary providing a mapping of package to directory names.
+
+``requires``
+
+   .. warning::
+      ``requires`` is superseded by ``install_requires`` and should not be used
+      anymore.
+
+``obsoletes``
+
+   .. warning::
+      ``obsoletes`` is not supported by ``pip`` and should not be used.
+
+``provides``
+
+   .. warning::
+      ``provides`` is not supported by ``pip`` and should not be used.
```

Comment:
This doesn't explain what these keywords *do*, and I also disagree with the idea that they "should not be used".

`provides` and `obsoletes` are actually valid keywords, they translate to [`Provides-Dist`](https://packaging.python.org/specifications/core-metadata/#provides-dist-multiple-use) and [`Obsoletes-Dist`](https://packaging.python.org/specifications/core-metadata/#obsoletes-dist-multiple-use), respectively.

It is true that `pip` does nothing with them, but this is very distinct from `requires`, which is actually deprecated in favor of `install_requires`. If these fields are included, what they do should be explained, possibly with a note that `pip` will not automatically do anything with this metadata.